src/mixins/nft.js

```diff
@@ -256,39 +261,33 @@ export default {
     },
     async updateNFTHistory() {
       this.isHistoryInfoLoading = true;
-      let hasHistoryInDatabase = false;
+      const historyOnChain = await this.getNFTEventsAll();
+      let history = historyOnChain;
+
       if (this.isWritingNFT) {
         try {
           const { data } = await this.$api.get(
             getNFTHistory({ classId: this.classId })
           );
-          this.NFTHistory = data.list;
-          hasHistoryInDatabase = true;
+          const historyInDB = data.list;
+          const eventMap = new Map();
+          historyOnChain
+            .filter(
+              e =>
+                !['new_class', 'mint_nft'].includes(e.event) &&
+                e.toWallet !== LIKECOIN_NFT_API_WALLET
+            )
+            .forEach(e => eventMap.set(`${e.txHash}-${e.nftId}`, e));
+          // overwrite on-chain event with db event
+          historyInDB.forEach(e => eventMap.set(`${e.txHash}-${e.nftId}`, e));
```

Comment:
I think we only need price info, can we only inject price info for an hash-id that already exists?